hipe4ml/analysis_utils.py

```diff
@@ -181,7 +185,84 @@ def train_test_generator(data_list, labels_list, sliced_df=False, **kwds):
         del data_frame
         df_tot_train_test = pd.concat(df_list, sort=True)
         del df_list
-        train_test = train_test_split(df_tot_train_test, np.array(labels_train_test), **kwds)
+        train_test = train_test_split(
+            df_tot_train_test, np.array(labels_train_test), **kwds)
         train_test[1], train_test[2] = train_test[2], train_test[1]
         train_test_slices.append(train_test)
     return train_test_slices
+
+
+def get_handler_from_large_file(file_name, tree_name, preselection=None, model_handler=None,
+                                score_cut=None, output_margin=True, max_workers=None):
+    """
+    Read a ROOT.TTree in different lazy chuncks. Chuncks are read sequentially or in parallel
+    and eventually pre-selections or ML selections are applied. This allows to preserve the
+    memory usage and speed-up the reading. Chuncks size is decided automatically
+
+    Parameters
+    -----------------------------------------------
+    file_name: str or list of str
+        Name of the input file where the data sit or list of input files
+
+    tree_name: str
+        Name of the tree within the input file, must be the same for all files
+
+    preselection: str
+        String containing the cuts to be applied as preselection on the data contained in the original
+        tree. The string syntax is the one required in the pandas.DataFrame.query() method.
+        You can refer to variables in the environment by prefixing them with an ‘@’ character like @a + b
+
+    model_handler: hipe4ml ModelHandler
+        Model handler to be applied as a preselection on the data contained in the original
+        tree. A column named model_output is added to the tree_handler
+
+    score_cut: int or float
+        Score to be applied as a starting preselection on the data contained in the original tree
```

Comment:
This does not support multiclass classifications. Maybe the code could be refactored adding another method for application on large files